File: wordle/globle.js

const fs = require("fs")

async function getGlobleAnswer() {
  const today = new Date().toISOString().split('T')[0];
  const listLength = 197;
  const response = await fetch(`https://globle-game.com/answer?day=${today}&list=${listLength}`);
  
  if (!response.ok) {
    throw new Error(`HTTP error! Status: ${response.status}`);
  }
  
  const data = await response.json();
  return data.answer;
}

decrypted = getGlobleAnswer()
  .then(answer => {
    fs.writeFileSync("C:/Users/pmcgi/OneDrive/Desktop/Visual Studio/wordle/globle.txt", answer);
  })
  .catch(error => {console.error('Error fetching the Globle answer:', error)});

File: wordle/globle.py
import requests
import datetime
import re
import subprocess
import json as j
import easygui

url = "https://globle-game.com/assets/index-DYefhQCl.js"
response = requests.get(url)
response.raise_for_status()
js_code = response.text
json = js_code.split("Mn=JSON.parse(`")[1].split("`),W")[0]
countries = j.loads(json)
listLength = len(countries)
url = "https://globle-game.com/assets/Game-D9rMaLVn.js"
response = requests.get(url)
js_code = response.text
password = js_code.split('const it="')[1].split('"')[0]
subprocess.run('"C:/Program Files/nodejs/node.exe" "C:/Users/pmcgi/OneDrive/Desktop/JavaScript/globle/globle.js"')
import decrypt
with open("C:/Users/pmcgi/OneDrive/Desktop/Visual Studio/wordle/globle.txt", "r") as f:
    encoded = f.read()
    f.close()
index = int(decrypt.decrypt(encoded, password))
easygui.textbox("Answer:",text=countries[index]["properties"]["NAME"])